app/assets/javascripts/modules/track-click.js

```diff
@@ -10,14 +10,24 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
       var options = {},
           category = element.attr('data-track-category'),
           action = element.attr('data-track-action'),
-          label = element.attr('data-track-label');
+          label = element.attr('data-track-label'),
+          dimension = element.attr('data-track-dimension'),
+          customDimension = element.data('track-custom-dimension');
 
       if (label) {
         options.label = label;
       }
 
       function trackClick() {
-        if (GOVUK.analytics && GOVUK.analytics.trackEvent) {
+        if (!GOVUK.analytics) {
+          return;
+        }
+
+        if (customDimension && dimension && GOVUK.analytics.setDimension) {
+          GOVUK.analytics.setDimension(customDimension, dimension);
```

Comment:
Rather than setting the dimension I think we should pass it as an option into the event itself:

See: https://github.com/alphagov/govuk_frontend_toolkit/blob/c4acf649030d8a7be1e1533117d36f9bebbe038b/javascripts/govuk/analytics/analytics.js#L37-L45

And:
https://developers.google.com/analytics/devguides/collection/analyticsjs/custom-dims-mets

eg
```
GOVUK.analytics.trackEvent(category, action, {dimension29: "dimensionValue"});
```

That _should_ work.